ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java

```diff
@@ -1162,15 +1249,21 @@ private void checkRouterExceptionPipeline(String expectedExceptionMsg, RestReque
   /**
    * Generate a {@link JSONObject} with a range header from a {@link ByteRange}
    * @param range the {@link ByteRange} to include in the headers.
+   * @param getOption the options to use while getting the blob.
    * @return the {@link JSONObject} with a range header, or null if {@code range} is null.
    * @throws Exception
    */
-  private JSONObject createRequestHeaders(ByteRange range) throws Exception {
-    if (range == null) {
+  private JSONObject createRequestHeaders(ByteRange range, GetOption getOption) throws Exception {
+    if (range == null && getOption == null) {
       return null;
```

Comment:
Either works I think i.e. one is not more correct than the other. Perhaps this is to avoid creating new empty objects that serve no purpose.